packages/cdk8s-cli/lib/import/type-generator.ts

```diff
@@ -214,6 +214,11 @@ export class TypeGenerator {
         if (propName.startsWith('x-')) {
           continue; // skip extensions for now
         }
+
+        if (propName.includes('_')) {
```

Comment:
Can you explain a bit more why this doesn't work with JSII? Are escape hatches supported in `cdk8s` yet?

Reply:
jsii requires that names will be capitalized in a certain way in order to be able to case-convert them to different languages in an idiomatic way. We might be able to lift off that restriction, but will require a bit more investigation. @romainmuller any thoughts?

We don't have escape hatches yet (#144)

Reply:
Yep. I'd like to allow more freedom in naming but right now we rely on tight constraints to make things work. Lifting the restriction will actually take a significant effort so it will be a longer haul kind of thing.

Basically the way I think about it is that it's come as a benefit of moving name generation (for target languages) earlier in the value flow (compiler pre-names everything, instead of code generators doing this at the last minute).